src/config.js

```diff
@@ -257,13 +257,20 @@ export default class Config {
    */
 
   getOfflineMirrorPath(tarUrl: ?string): ?string {
-    const registry = this.registries.npm;
-    if (registry == null) {
-      return null;
+    //
+    let mirrorPath;
+
+    for (const key of Object.keys(this.registries)) {
+      const registry = this.registries[key];
+      const registryMirrorPath = registry.config['yarn-offline-mirror'];
+
+      if (registryMirrorPath == null) {
+        continue;
```

Comment:
This code will return value from `yarn-offline-mirror` of the last key in this.registries iteration.
Afaik we don't get guarantee of iteration of Object keys so this may be random value from .yarnrc or .npmrc.

Should you just get this.registries.yarn and this.registries.npm explicitly?
And then order the priority manually yarn > npm.